File: clients/android/app/src/main/java/tv/plurx/app/data/Session.kt
package tv.plurx.app.data

/**
 * Live connection state, read by the OkHttp auth interceptor (and the image /
 * player data sources) on every request. Set once at connect and after login;
 * kept as a plain holder so a single OkHttpClient serves the whole app and the
 * token can change without rebuilding it.
 */
object Session {
    /** Server origin, no trailing slash, e.g. `http://192.168.1.10:32600`. */
    @Volatile
    var origin: String = ""

    /** Bearer token, or null when signed out. */
    @Volatile
    var token: String? = null

    /** Absolute URL for a server-relative path (`/api/v1/images/…`). */
    fun url(path: String): String =
        if (path.startsWith("http")) path else origin + path
}

File: clients/apple/Sources/Session.swift
import Foundation

/// Live connection state. The origin + token are set once at connect/login and
/// read wherever a request, image, or media URL is built. Two auth shapes:
/// API/image requests carry `Authorization: Bearer`, while AVPlayer URLs (which
/// can't set headers) carry the token inline as `?token=` — both accepted by
/// the server's `AuthUser` extractor.
final class Session: @unchecked Sendable {
    static let shared = Session()

    /// Server origin, no trailing slash, e.g. `http://192.168.1.10:32600`.
    var origin: String = ""
    /// Bearer token, or nil when signed out.
    var token: String?

    /// Absolute URL for a server-relative path.
    func url(_ path: String) -> URL? {
        if path.hasPrefix("http") { return URL(string: path) }
        return URL(string: origin + path)
    }

    /// Absolute URL with the token inline — for AVPlayer / `<img>`-style loads
    /// that can't set an Authorization header. Capability-authed HLS playlists
    /// (which already carry an unguessable session id) don't need this.
    func mediaURL(_ path: String) -> URL? {
        guard let base = url(path) else { return nil }
        guard let token, !path.hasPrefix("http") else { return base }
        var comps = URLComponents(url: base, resolvingAgainstBaseURL: false)
        var items = comps?.queryItems ?? []
        items.append(URLQueryItem(name: "token", value: token))
        comps?.queryItems = items
        return comps?.url ?? base
    }

    /// Add the bearer header to an API/image request.
    func authorize(_ request: inout URLRequest) {
        if let token {
            request.setValue("Bearer \(token)", forHTTPHeaderField: "Authorization")
        }
    }
}
